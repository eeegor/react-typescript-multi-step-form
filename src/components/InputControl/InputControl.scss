@import '../../variables';

.input-control__label {
	font-size: 1.5rem;
	display: block;
	margin-bottom: 2rem;
	text-align: center;
}

.input-control__info,
.input-control__errors {
	margin-bottom: 1rem;
	text-align: center;
}

.input-control--danger {
	position: relative;

	.input-control__errors {
		position: absolute;
		top: 2.5rem;
		left: 0;
		right: 0;
		color: $color-danger;
		margin-bottom: 0.5rem;
	}
	input {
		border-color: $color-danger;
	}
}

.input-control--success {
	input {
		border-color: $color-success;
	}
}
