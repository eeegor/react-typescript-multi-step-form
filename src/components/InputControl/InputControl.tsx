/**
 * Component InputControl
 */

import * as React from 'react';
import * as classnames from 'classnames';
import './InputControl.scss';

export interface InputControlProps {
	id: string;
	className?: string;
	required?: boolean;
	type?: 'text' | 'email' | 'radio' | 'select-radio' | 'submit';
	status?: 'success' | 'danger';
}

export class InputControl extends React.Component<InputControlProps> {
	render(): JSX.Element {
		const { id, className, children, required, type, status } = this.props;
		const cssClass = classnames(
			`input-control input-control-${type || 'text'}`,
			status && `input-control--${status}`,
			required && `input-control--${required}`,
			className
		);

		return (
			<div id={`input-control-${id}`} className={cssClass}>
				{children}
			</div>
		);
	}
}
