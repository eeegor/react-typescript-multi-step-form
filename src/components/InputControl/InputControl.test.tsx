import React from 'react';
import renderer from 'react-test-renderer';
import { InputControl } from './InputControl';
import { mount, shallow } from 'enzyme';

describe('InputControl', () => {
	it('renders markup correctly', () => {
		const component = <InputControl id="a-unique-id" />;
		const tree = renderer.create(component).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should adopt a status class from attribute', () => {
		const wrapper = shallow(<InputControl id="a-unique-id" status="success" />);
		expect(wrapper.hasClass('input-control--success')).toEqual(true);
	});

	it('should adopt a valid class if valid', () => {
		const wrapper = shallow(<InputControl id="a-unique-id" valid={true} />);
		expect(wrapper.hasClass('input-control--success')).toEqual(true);
	});

	it('should adopt a status danger class if has errors', () => {
		const wrapper = shallow(<InputControl id="a-unique-id" errors={'Error'} />);
		expect(wrapper.hasClass('input-control--danger')).toEqual(true);
	});

	it('should adopt a required class from attribute', () => {
		const wrapper = shallow(<InputControl id="a-unique-id" required={true} />);
		expect(wrapper.hasClass('input-control--required')).toEqual(true);
	});
});
