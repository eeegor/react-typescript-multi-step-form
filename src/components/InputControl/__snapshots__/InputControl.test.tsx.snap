// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputControl renders markup correctly 1`] = `
<div
  className="input-control input-control-text"
  id="input-control-a-unique-id"
/>
`;
